js/views/userPage/UserPage.js

```diff
@@ -18,12 +20,16 @@ export default class extends baseVw {
     super(options);
     this.options = options;
 
+    this.ownPage = this.model.id === app.profile.id;
+
     this.state = options.state || 'store';
+    if (this.state === 'store' && !this.ownPage && !this.model.get('vendor')) {
```

Comment:
I saw the note in the description about some complications to this approach. If I'm missing something... please let me know.